crates/test-programs/tests/wasi-sockets.rs

```diff
@@ -0,0 +1,93 @@
+#![cfg(all(feature = "test_programs", not(skip_wasi_sockets_tests)))]
+use cap_std::ambient_authority;
+use wasmtime::component::Linker;
+use wasmtime::{Config, Engine, Store};
+use wasmtime_wasi::preview2::{self, command::Command, Table, WasiCtx, WasiCtxBuilder, WasiView};
+
+lazy_static::lazy_static! {
+    static ref ENGINE: Engine = {
+        let mut config = Config::new();
+        config.wasm_backtrace_details(wasmtime::WasmBacktraceDetails::Enable);
+        config.wasm_component_model(true);
+        config.async_support(true);
+
+        let engine = Engine::new(&config).unwrap();
+        engine
+    };
+}
+// uses ENGINE, creates a fn get_component(&str) -> Component
+include!(concat!(
+    env!("OUT_DIR"),
+    "/wasi_sockets_tests_components.rs"
+));
+
+struct SocketsCtx {
+    table: Table,
+    wasi: WasiCtx,
+}
+
+impl WasiView for SocketsCtx {
+    fn table(&self) -> &Table {
+        &self.table
+    }
+    fn table_mut(&mut self) -> &mut Table {
+        &mut self.table
+    }
+    fn ctx(&self) -> &WasiCtx {
+        &self.wasi
+    }
+    fn ctx_mut(&mut self) -> &mut WasiCtx {
+        &mut self.wasi
+    }
+}
+
+async fn run(name: &str) -> anyhow::Result<()> {
+    let component = get_component(name);
+    let mut linker = Linker::new(&ENGINE);
+
+    preview2::bindings::io::streams::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::poll::poll::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::exit::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::stdin::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::stdout::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::stderr::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::terminal_input::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::terminal_output::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::terminal_stdin::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::terminal_stdout::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::terminal_stderr::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::cli::environment::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::filesystem::types::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::filesystem::preopens::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::sockets::tcp::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::sockets::tcp_create_socket::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::sockets::network::add_to_linker(&mut linker, |x| x)?;
+    preview2::bindings::sockets::instance_network::add_to_linker(&mut linker, |x| x)?;
```

Comment:
rather than listing these out explicitly, can you use `command::add_to_linker` to get them all?

Reply:
Done